slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -285,7 +321,7 @@ std::string SLAMServiceImpl::GetNextDataFile() {
 
 void SLAMServiceImpl::CreateMap() {
     cartographer::mapping::TrajectoryBuilderInterface *trajectory_builder;
-    int trajectory_id;
```

Comment:
I think we should keep this here, and for the global latest pose to be updated within this function. The reason for not making `trajectory_id` one global variable within the slam service is that `trajectory_id` is used ubiquitously everywhere within cartographer. I am not sure that we won't be juggling multiple `trajectory_ids` in the future, which is why I'd like for them to be kept as a local variable within functions for now.